source/features/clean-repo-filelist-actions.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'dom-chef';
+import onetime from 'onetime';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+import {PlusIcon, SearchIcon} from '@primer/octicons-react';
+
+import {wrap} from '../helpers/dom-utils';
+import features from '.';
+
+function init(): void {
+	// Only select desktop version of the button
+	observe('.d-flex > [data-hotkey="t"]:not(.rgh-clean-actions)', {
+		add(searchButton) {
+			searchButton.classList.add('tooltipped', 'tooltipped-ne', 'rgh-clean-actions');
+			searchButton.setAttribute('aria-label', 'Go to file');
+
+			searchButton.firstChild!.replaceWith(<SearchIcon/>);
+
+			// Exclude logged out, mobile or file pages
+			const addButtonText = searchButton.nextElementSibling?.querySelector('.d-md-flex');
+			if (addButtonText) {
+				addButtonText.classList.replace('d-md-flex', 'd-md-block');
+				addButtonText.firstChild!.replaceWith(<PlusIcon/>);
+				wrap(addButtonText.parentElement!.parentElement!, <div className="tooltipped tooltipped-ne" aria-label="Add file"/>);
+			}
+		}
+	});
+
+	observe('get-repo summary:not(.rgh-clean-actions)', {
+		add(button) {
+			button.classList.add('rgh-clean-actions');
+			button.firstElementChild!.nextSibling!.remove();
```

Comment:
```suggestion
			button.firstElementChild!.nextSibling!.remove(); // What is this removing?
```

Reply:
The "Code" text node in the `<summary>` of the actions dropdown. (The text nodes of the other buttons are directly replaced by icons).

Reply:
I think so. This whole block is silent, no comment anywhere and the variable is simply button. 

It’s worth explaining in both cases why the details is being wrapped like that, with an identical comment. Otherwise we’ll forget why we do such weird things.